### ISSUE TITLE
TimeBasedRollingFileAppender daily rolling log file creation ignores timezone in 1.2.0

### ISSUE THREAD

**Issue by M1chae1** [log4cplus/log4cplus#207]
I found a problem that the new daily log file will be created on UTC 00:00 but not localtime 00:00

PC's Timezone: UTC+08:00
Version: latest version of branch 1.2.x
Build: Visual Studio 2012 x64

I have written a simple console program to test about it. There is the source code

```
#include <iostream>

#include "log4cplus/logger.h"
#include "log4cplus/loggingmacros.h"
#include "log4cplus/configurator.h"
#include "log4cplus/fileappender.h"

#pragma comment(lib, "log4cplusU.lib")

int _tmain(int argc, _TCHAR* argv[])
{
    log4cplus::initialize();
    log4cplus::PropertyConfigurator::doConfigure(LOG4CPLUS_TEXT("log4cplus.properties"));
    log4cplus::Logger logger = log4cplus::Logger::getInstance(LOG4CPLUS_TEXT("TEST_DAILY_ROLLING"));

    char c = ' ';

    while ('x' != c)
    {
        std::cout << "Press any key to write log or 'x' to exit";
        c = getchar();

        if (c != 'x')
        {
            LOG4CPLUS_DEBUG(logger, "Debug log message");
        }
    }

    return 0;
}
```

There is the content of log4cplus.properties

```
log4cplus.logger.TEST_DAILY_ROLLING = DEBUG, TestLogAppender

#TestLogAppender Configuration
log4cplus.appender.TestLogAppender = log4cplus::TimeBasedRollingFileAppender
log4cplus.appender.TestLogAppender.FilenamePattern = %d{yyyyMMdd}.log
log4cplus.appender.TestLogAppender.MaxHistory = 365
log4cplus.appender.TestLogAppender.Schedule = DAILY
log4cplus.appender.TestLogAppender.RollOnClose = false
log4cplus.appender.TestLogAppender.layout = log4cplus::PatternLayout
log4cplus.appender.TestLogAppender.layout.ConversionPattern = %D{%Y-%m-%d %H:%M:%S.%q} [%-5p] %m%n
```

Test Procedure
1. Execute the exe file
2. Press Enter to write log
3. Change the time to any time within 00:00:00 - 07:59:59 on next day. i.e. my timezone is UTC +08:00
4. Press Enter to write log. The log message will be written into yesterday log file
5. Change the time to after 08:00. i.e. my timezone is UTC +08:00
6. Press Enter to write log. The new daily log file is created.

There is the partial result of the log file generated

20161027.log

```
......
2016-10-28 07:59:47.160 [DEBUG] Debug log message
2016-10-28 07:59:59.873 [DEBUG] Debug log message
```

20161028.log

```
2016-10-28 08:00:00.045 [DEBUG] Debug log message
2016-10-28 08:00:00.264 [DEBUG] Debug log message
......
```

If the program is opened after the day change, it works properly. However, it doesn't work while the program is running over midnight.

The bug #136 is closed but the bug is still exists.

**Comment by wilx** [comment]
@cfyzium What do you think about https://github.com/wilx/log4cplus/commit/cf26336a936495e088b52045c8bb7c964b71c92e for a fix for this issue?

**Comment by wilx** [comment]
I believe this should be fixed now on 1.2.x branch by fixes for #136.
